nevergrad/optimization/test_optimizerlib.py

```diff
@@ -496,11 +496,79 @@ def test_bo_ordering() -> None:
 
 
 @pytest.mark.parametrize(  # type: ignore
-    "name,expected", [("NGOpt2", ["TBPSA", "CMandAS2"])]
+    "name,budget,expected", [
+        ("NGOpt8", 100, ["OnePlusOne", "OnePlusOne"]),
+        ("NGOpt8", 200, ["SQP", "SQP"])
+        ]
 )
-def test_ngo_split_optimizer(name: str, expected: tp.List[str]) -> None:
+def test_ngo_split_optimizer(name: str, budget: int, expected: tp.List[str]) -> None:
     param = ng.p.Choice(["const", ng.p.Array(init=[1, 2, 3])])
     Opt = optlib.registry[name]
-    opt = optlib.ConfSplitOptimizer(multivariate_optimizer=Opt)(param, budget=1000)
+    opt = optlib.ConfSplitOptimizer(multivariate_optimizer=Opt)(param, budget=budget)
     names = [o.optim.name for o in opt.optims]  # type: ignore
     assert names == expected
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "name,expected", [("NGOpt8", ['SQP', 'monovariate', 'monovariate'])]
+)
+def test_instrumented_ngo_split_optimizer(name: str, expected: tp.List[str]) -> None:
+    param = ng.p.Instrumentation(
+      # a log-distributed scalar between 0.001 and 1.0
+      learning_rate=ng.p.Log(lower=0.001, upper=1.0),
+      # an integer from 1 to 12
+      batch_size=ng.p.Scalar(lower=1, upper=12).set_integer_casting(),
+      # either "conv" or "fc"
+      architecture=ng.p.Choice(["conv", "fc"])
+    )
+    Opt = optlib.registry[name]
+    opt = optlib.ConfSplitOptimizer(multivariate_optimizer=Opt)(param, budget=1000)
+    names = [o.optim.name if o.dimension != 1 else "monovariate" for o in opt.optims]  # type: ignore
+    assert names == expected
+
+
+
+@pytest.mark.parametrize("budget", [100, 300, 1000, 3000])  # type: ignore
+def test_ngopt_on_simple_realistic_scenario(budget: int) -> None:
+    def fake_training(learning_rate: float, batch_size: int, architecture: str) -> float:
+      # optimal for learning_rate=0.2, batch_size=4, architecture="conv"
+      return (learning_rate - 0.2)**2 + (batch_size - 4)**2 + (0 if architecture == "conv" else 10)
+    
+    # Instrumentation class is used for functions with multiple inputs
+    # (positional and/or keywords)
+    parametrization = ng.p.Instrumentation(
+      # a log-distributed scalar between 0.001 and 1.0
+      learning_rate=ng.p.Log(lower=0.001, upper=1.0),
+      # an integer from 1 to 12
+      batch_size=ng.p.Scalar(lower=1, upper=12).set_integer_casting(),
+      # either "conv" or "fc"
+      architecture=ng.p.Choice(["conv", "fc"])
+    )
+ 
+    optimizer = ng.optimizers.NGOpt(parametrization=parametrization, budget=budget)
+    recommendation = optimizer.minimize(fake_training)
+    assert fake_training(**recommendation.kwargs) < 5e-2
+
+
+
+@pytest.mark.parametrize("budget", [100, 300, 1000, 3000])  # type: ignore
+def test_ngopt_on_continuous_realistic_scenario(budget: int) -> None:
```

Comment:
please merge with the function above, they seem very similar, just add a `param_name` parameter

Reply:
Ok done.